filter-rspamd.go

```diff
@@ -508,7 +508,11 @@ func trigger(actions map[string]func(*session, []string), atoms []string) {
 		sessions[s.id] = &s
 	}
 
-	s := sessions[atoms[5]]
+	s, ok := sessions[atoms[5]]
```

Comment:
actually that's a good catch.

As it stands right now, opensmtpd terminates when the filter terminates. I wonder if taking down the mail server is the best possible behavior when an unknown session id turns up...

Reply:
It's indeed not nice to have a production server taken down. On the other hand, that's how the filter behaves in many other situations when it receives invalid input. One could argue that this will never happen unless there's a severe bug in the protocol code and shutting down the server will increase visibility of such severe issues.

If we keep running the server with broken protocol code, this might potentially lead to emails being accepted and then dropped (and similar issues), which is arguably even worse than getting temporary delivery failures.

Reply:
@jdelic if an unknown session id turns up, you have more serious issues as OpenSMTPD will never generate bad input to the filters. this would mean that there's a severe memory corruption and keeping it running may cause mail loss, your best option is to abord as soon as possible.

@lfos exactly